app/context.go

```diff
@@ -0,0 +1,100 @@
+package app
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+type ContextKey string
+
+const (
+	TraceIDKey  ContextKey = "trace_id"
+	UserIDKey   ContextKey = "user_id"
+	TenantIDKey ContextKey = "tenant_id"
+)
+
+// Context carries the context of the current execution.
+type Context struct {
+	// original context
+	context.Context
+}
+
+// UserID returns the user id
+func (sc *Context) UserID() (string, bool) {
+	userID := sc.Context.Value(UserIDKey)
+
+	if id, ok := userID.(string); ok {
+		return id, ok
```

Comment:
If ok is true, usually `return id, true` would be written for readability

The same way we usually write `return nil` while `return err` would be the sane in this example

```go
if err != nil {
     return err
}

return nil
```

But as your code is simple the code could be  refactored to be

```go
id, ok := userID.(string)

return id, ok
```

As id default to empty string when not found.

I'm unsure (I'm on phone now) if code could be written directly as this

```go
return userID.(string)
```

Reply:
Done 👍  unfortunately the `userID.(string)` does not compile